django/utils/functional.py

```diff
@@ -94,7 +94,7 @@ def __prepare_class__(cls):
                 if six.PY3:
                     cls.__str__ = cls.__text_cast
                 else:
-                    cls.__unicode__ = cls.__text_cast
+                    cls.__unicode__ = cls.__str__ = cls.__text_cast
```

Comment:
`__str__` and `__repr__` methods must return bytestring in Python 2. See https://docs.python.org/2/reference/datamodel.html#object.__str__